config/samples/sample-app-agent.yml

```diff
@@ -0,0 +1,132 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: quarkus-test-agent
+    app.kubernetes.io/component: quarkus-test-agent
+    app.kubernetes.io/instance: quarkus-test-agent
+  name: quarkus-test-agent
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      deployment: quarkus-test-agent
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        openshift.io/generated-by: OpenShiftNewApp
```

Comment:
I think we can remove this annotation. 

How about this? This should give a minimal specs for the deployment and service objects.

```diff
diff --git a/config/samples/sample-app-agent.yml b/config/samples/sample-app-agent.yml
index 140ba5a..5e2acd3 100644
--- a/config/samples/sample-app-agent.yml
+++ b/config/samples/sample-app-agent.yml
@@ -4,28 +4,16 @@ kind: Deployment
 metadata:
   labels:
     app: quarkus-test-agent
-    app.kubernetes.io/component: quarkus-test-agent
-    app.kubernetes.io/instance: quarkus-test-agent
   name: quarkus-test-agent
 spec:
-  progressDeadlineSeconds: 600
   replicas: 1
-  revisionHistoryLimit: 10
   selector:
     matchLabels:
-      deployment: quarkus-test-agent
-  strategy:
-    rollingUpdate:
-      maxSurge: 25%
-      maxUnavailable: 25%
-    type: RollingUpdate
+      app: quarkus-test-agent
   template:
     metadata:
-      annotations:
-        openshift.io/generated-by: OpenShiftNewApp
-      creationTimestamp: null
       labels:
-        deployment: quarkus-test-agent
+        app: quarkus-test-agent
     spec:
       containers:
       - env:
@@ -74,18 +62,13 @@ spec:
           capabilities:
             drop:
             - ALL
-        terminationMessagePath: /dev/termination-log
-        terminationMessagePolicy: File
         volumeMounts:
         - mountPath: /var/run/secrets/myapp/truststore.p12
           name: truststore
           subPath: truststore.p12
-      dnsPolicy: ClusterFirst
       restartPolicy: Always
-      schedulerName: default-scheduler
       securityContext:
         runAsNonRoot: true
-      terminationGracePeriodSeconds: 30
       volumes:
       - name: truststore
         secret:
@@ -97,14 +80,10 @@ kind: Service
 metadata:
   labels:
     app: quarkus-test-agent
-    app.kubernetes.io/component: quarkus-test-agent
-    app.kubernetes.io/instance: quarkus-test-agent
   name: quarkus-test-agent
 spec:
-  internalTrafficPolicy: Cluster
-  ipFamilies:
-  - IPv4
-  ipFamilyPolicy: SingleStack
+  selector:
+    app: quarkus-test-agent
   ports:
   - name: jfr-jmx
     port: 9097
@@ -126,7 +105,3 @@ spec:
     port: 8443
     protocol: TCP
     targetPort: 8443
-  selector:
-    deployment: quarkus-test-agent
-  sessionAffinity: None
-  type: ClusterIP

```